pkgs/development/compilers/gnatboot/default.nix

```diff
@@ -1,55 +1,51 @@
-{ lib, stdenv, fetchurl }:
+{ stdenv, lib, autoPatchelfHook, fetchzip, lzma, ncurses5, readline, gmp, mpfr
+, expat, libipt, zlib, dejagnu, sourceHighlight, python3, elfutils, guile, glibc
+}:
 
-if stdenv.hostPlatform != stdenv.targetPlatform
-then builtins.throw "gnatboot can't cross-compile"
-else
+stdenv.mkDerivation rec {
+  pname = "gnatboot";
+  version = "11.2.0-4";
 
-stdenv.mkDerivation {
-  pname = "gentoo-gnatboot";
-  version = "4.1";
-
-  src = if stdenv.hostPlatform.system == "i686-linux" then
-    fetchurl {
-      url = "mirror://gentoo/distfiles/gnatboot-4.1-i386.tar.bz2";
-      sha256 = "0665zk71598204bf521vw68i5y6ccqarq9fcxsqp7ccgycb4lysr";
-    }
-  else if stdenv.hostPlatform.system == "x86_64-linux" then
-    fetchurl {
-      url = "mirror://gentoo/distfiles/gnatboot-4.1-amd64.tar.bz2";
-      sha256 = "1li4d52lmbnfs6llcshlbqyik2q2q4bvpir0f7n38nagp0h6j0d4";
-    }
-  else
-    throw "Platform not supported";
+  src = fetchzip {
+    url = "https://github.com/alire-project/GNAT-FSF-builds/releases/download/gnat-${version}/gnat-x86_64-linux-${version}.tar.gz";
+    hash = "sha256-8fMBJp6igH+Md5jE4LMubDmC4GLt4A+bZG/Xcz2LAJQ=";
+  };
 
-  dontStrip = 1;
+  propagatedNativeBuildInputs = [
```

Comment:
Why are these all propagated? For the previous gnatboot it didn't seem necessary to propagate them, right?

Reply:
autoPatchElfHook only operates on this derivation right? I'd expect those deps in nativeBuildInputs should be enough if they're all patched into the RPATH

Reply:
I tried that in the initial testing of it before I sent this PR in and it didn't work properly for some reason. I altered `default.nix` as suggested and it seems to work this time... Not sure what changed.